src/utils/utils_log.c

```diff
@@ -0,0 +1,222 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+#ifndef __WIN32
+#define _GNU_SOURCE 1
+#endif
+#include <stdarg.h>
+#include <string.h>
+#include <time.h>
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#include "utils_common.h"
+#include "utils_log.h"
+
+typedef struct {
+    int timestamp;
+    int pid;
+    utils_log_level_t level;
+    utils_log_level_t flushLevel;
+    FILE *output;
+} utils_log_config;
+
+utils_log_config loggerConfig = {0, 0, LOG_ERROR, LOG_ERROR, NULL};
+
+static const char *level_to_str(utils_log_level_t l) {
+    switch (l) {
+    case LOG_DEBUG:
+        return "DEBUG";
+    case LOG_ERROR:
+        return "ERROR";
+    case LOG_INFO:
+        return "INFO";
+    case LOG_WARNING:
+        return "WARN";
+    default:
+        ASSERT(0);
+        return "";
+    }
+}
+static const char *match_env_var(const char *envVar, const char *prefix,
```

Comment:
please add an empty line before function definition

Reply:
Line added- 

And #define _GNU_SOURCE 1 affects others headers - it is common practice to define it as fist.